### PR TITLE
Get the last time when the tx write buffer got emptied

### PR DESCRIPTION
**Pull request by amilcarlucas**
```
Binary Name      Text [B]        Data [B]     BSS (B)        Total Flash Change [B] (%)      Flash Free After PR (B)
---------------  --------------  -----------  -------------  ----------------------------  -------------------------
ardurover        108 (+0.0066%)  0 (0.0000%)  -4 (-0.0015%)  108 (+0.0066%)                                   337492
blimp            108 (+0.0081%)  0 (0.0000%)  4 (+0.0015%)   108 (+0.0080%)                                   623048
arducopter       108 (+0.0061%)  0 (0.0000%)  -4 (-0.0015%)  108 (+0.0061%)                                   193140
arduplane        108 (+0.0061%)  0 (0.0000%)  -4 (-0.0015%)  108 (+0.0061%)                                   201456
ardusub          108 (+0.0069%)  0 (0.0000%)  4 (+0.0015%)   108 (+0.0069%)                                   400760
antennatracker   108 (+0.0082%)  0 (0.0000%)  -4 (-0.0015%)  108 (+0.0082%)                                   645924
arducopter-heli  108 (+0.0061%)  0 (0.0000%)  -4 (-0.0015%)  108 (+0.0061%)                                   186972
```

### CODE REVIEW

File: libraries/AP_HAL_ChibiOS/UARTDriver.h [hunk 1]
```diff
@@ -144,6 +144,9 @@ class ChibiOS::UARTDriver : public AP_HAL::UARTDriver {
      */
     bool is_dma_enabled() const override { return rx_dma_enabled && tx_dma_enabled; }
 
+    // get the last time when the tx write buffer got emptied
+    uint32_t get_last_tx_empty_us() const override { return (tx_dma_enabled && _writebuf.is_empty()) ? _last_write_completed_us : 0; }
```
Comment (khancyr) [inline libraries/AP_HAL_ChibiOS/UARTDriver.h:148]: why this depends on DMA config ?

Comment (amilcarlucas) [reply libraries/AP_HAL_ChibiOS/UARTDriver.h:148]: Because the _last_write_completed_us only gets written to if tx_dma_enabled == true

**Comment by khancyr** [comment]
@amilcarlucas do you still need this ?

**Comment by amilcarlucas** [comment]
Size comparison added on the description

**Comment by amilcarlucas** [comment]
I just realized that FETtec no longer needs this, closing

**Pull request closed without merging**